Sources/Core/AWSSDKIdentity/Sources/AWSSDKIdentity/AWSCredentialIdentityResolvers/ProfileAWSCredentialIdentityResolver.swift

```diff
@@ -139,12 +144,21 @@ public struct ProfileAWSCredentialIdentityResolver: AWSCredentialIdentityResolve
             visitedProfiles.count == 1 &&
             profile.val(for: "source_profile") == profile.name {
             resolvers.append { identityProperties in
-                let access = profile.string(for: .init(stringLiteral: "aws_access_key_id"))!
-                let secret = profile.string(for: .init(stringLiteral: "aws_secret_access_key"))!
-                let accountID = profile.string(for: .init(stringLiteral: "aws_account_id"))
-                let sourceCreds = AWSCredentialIdentity(accessKey: access, secret: secret, accountID: accountID)
+                let sourceCreds = try await SharedConfigStaticAWSCredentialIdentityResolver(
+                    profileName: profile.name,
+                    configFilePath: configFilePath,
+                    credentialsFilePath: credentialsFilePath
+                ).getIdentity(identityProperties: identityProperties)
+                let sourceCredsWithUpdatedFeatureIDs = prependFeatureIDs(
+                    featureIDsToPrepend: credentialFeatureIDs + [
+                        CredentialFeatureID.CREDENTIALS_PROFILE_SOURCE_PROFILE.rawValue
+                    ],
+                    creds: sourceCreds
+                )
                 return try await STSAssumeRoleAWSCredentialIdentityResolver(
-                    awsCredentialIdentityResolver: StaticAWSCredentialIdentityResolver(sourceCreds),
+                    awsCredentialIdentityResolver: StaticAWSCredentialIdentityResolver(
+                        sourceCredsWithUpdatedFeatureIDs
+                    ),
```

Comment:
Credential feature IDs resolved so far get prepended to credential feature ID of newly resolved credentials. This is a common pattern for other places below.